src/Request/GetParcelShops.php

```diff
@@ -4,68 +4,36 @@
 
 namespace Answear\GlsBundle\Request;
 
-use Answear\GlsBundle\Enum\CountryCode;
-use Answear\GlsBundle\Enum\ParcelShopAction;
+use Answear\GlsBundle\Enum\CountryCodeEnum;
 
 class GetParcelShops implements RequestInterface
 {
-    private const ENDPOINT = 'psmap/psmap_getdata.php';
+    private const ENDPOINT = 'data/deliveryPoints/';
     private const HTTP_METHOD = 'GET';
 
-    private ParcelShopAction $action;
-    private ?CountryCode $countryCode;
-    private ?string $senderId;
-    private ?bool $pclShopIn;
-    private ?bool $parcelLockIn;
-    private ?bool $codHandler;
-    private ?bool $dropOff;
-    private ?string $shopId;
-
-    public function __construct(
-        ParcelShopAction $action,
-        ?CountryCode $countryCode = null,
-        ?string $senderId = null,
-        ?bool $withPclShopIn = null,
-        ?bool $withParcelLockIn = null,
-        ?bool $codHandler = null,
-        ?bool $dropOff = null,
-        ?string $shopId = null
-    ) {
-        $this->action = $action;
-        $this->countryCode = $countryCode;
-        $this->senderId = $senderId;
-        $this->pclShopIn = $withPclShopIn;
-        $this->parcelLockIn = $withParcelLockIn;
-        $this->codHandler = $codHandler;
-        $this->dropOff = $dropOff;
-        $this->shopId = $shopId;
+    public function __construct(public CountryCodeEnum $countryCode)
+    {
     }
 
     public function getEndpoint(): string
     {
-        return self::ENDPOINT;
+        return self::ENDPOINT . $this->getJsonFile();
     }
 
     public function getMethod(): string
     {
         return self::HTTP_METHOD;
     }
 
-    public function getUrlQuery(): ?string
+    private function getJsonFile(): string
     {
-        $urlQuery = http_build_query(
-            [
-                'action' => $this->action->getValue(),
-                'ctrcode' => null === $this->countryCode ? null : $this->countryCode->getValue(),
-                'senderid' => $this->senderId,
-                'pclshopin' => null === $this->pclShopIn ? null : (int) $this->pclShopIn,
-                'parcellockin' => null === $this->parcelLockIn ? null : (int) $this->parcelLockIn,
-                'codhandler' => null === $this->codHandler ? null : (int) $this->codHandler,
-                'dropoff' => null === $this->dropOff ? null : (int) $this->dropOff,
-                'pclshopid' => $this->shopId,
-            ]
-        );
-
-        return empty($urlQuery) ? null : $urlQuery;
+        return match ($this->countryCode) {
```

Comment:
seems to me better to do: `sprintf('%s.json',strtolower($this->countryCode->getValue()))`

if we need to handle another country then there is no need to change anything here

Reply:
Done